src/commands/src/helpers/dependency-helpers.ts

```diff
@@ -33,12 +34,47 @@ async function createEnvProdFile(repoDir: string, projectName: string, baseUrl:
   }
 }
 
-export async function prepareBuildAndGetDependencies(repoName: string, projectName: string, baseUrl: string) {
+export async function downloadBuildDependencies(buildDependencies: BuildDependencies, repoName: string) {
+  const commands: string[] = [];
+  await Promise.all(
+    buildDependencies.map(async (dependencyInfo) => {
+      try {
+        const repoDir = getRepoDir(repoName);
+        const targetDir = dependencyInfo.targetDir ?? repoDir;
+        await executeCliCommand({
+          command: 'mkdir',
+          args: ['-p', targetDir],
+        });
+        commands.push(`mkdir -p ${targetDir}`);
+        dependencyInfo.dependencies.map((dep) => {
```

Comment:
I think we will want to do an `await Promise.all()` here as well so that we ensure that these promises complete before continuing.